components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/ApplicationsApiServiceImpl.java

```diff
@@ -210,6 +211,16 @@ public Response applicationsApplicationIdPut(String applicationId, ApplicationDT
             }
 
             Application application = ApplicationMappingUtil.fromDTOtoApplication(body, username);
+            if (!ApplicationStatus.APPLICATION_APPROVED.equals(application.getStatus())) {
+                String errorMessage = "Application " + applicationId + " is not active";
+                ExceptionCodes exceptionCode = ExceptionCodes.APPLICATION_INACTIVE;
+                APIManagementException e = new APIManagementException(errorMessage, exceptionCode);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
```

Comment:
an empty paramList is passed. We can add the application id to the paramList I guess.